Templates/Templates/AutoDecodable.stencil

```diff
@@ -9,28 +11,85 @@ extension KeyedDecodingContainer {
 
 {# these are a giant line because of spacing, to debug just seperate the if into lines#}
 {% macro defaultValueFor variable %}{% set typename %}{{ variable.typeName.name }}{% endset %}{% if variable|annotated:"codingDefaultValue" %}{{ variable.annotations.codingDefaultValue }}){% elif variable.isOptional %}nil){% elif variable.isArray %}[]){% elif typename == "String" %}""){% elif typename == "Int" or typename == "Double" or typename == "Float" or typename == "NSNumber" %}0){% elif typename == "Bool" %}false){% endif %}{% endmacro %}
-{% macro nameWithTransform name transform %}{% if transform == "camelToSnake" or not transform %}{{ name|camelToSnakeCase }}{% elif transform == "camelToSnake:idToUUID" %}{{ name|camelToSnakeCase|replace:"id","uuid" }}{% elif transform == "idToUUID" %}{{ name|replace:"id","uuid" }}{% endif %}{% endmacro %}
-{% macro keyedNameFor type  variable %}{% if variable|annotated:"codingKey" %}{{ variable.annotations.codingKey }}{% elif type|annotated:"codingKeyTransform" %}{% call nameWithTransform variable.name type.annotations.codingKeyTransform %}{% else %}{{ variable.name }}{% endif %}{% endmacro %}
+{% macro decodeNameWithTransform name transform %}{% if transform == "camelToSnake" or not transform %}{{ name|camelToSnakeCase }}{% elif transform == "camelToSnake:idToUUID" %}{{ name|camelToSnakeCase|replace:"id","uuid" }}{% elif transform == "idToUUID" %}{{ name|replace:"id","uuid" }}{% endif %}{% endmacro %}
+{% macro keyedNameFor type  variable %}{% if variable|annotated:"codingKey" %} = "{{ variable.annotations.codingKey }}"{% elif type|annotated:"codingKeyTransform" %}= "{% call decodeNameWithTransform variable.name type.annotations.codingKeyTransform %}"{% endif %}{% endmacro %}
 
 {# the keys to use to decoding #}
 {% macro codingKeys type %}
-  fileprivate enum CodableKeys: String, CodingKey {
-    {% for variable in type.allVariables where variable|!annotated:"skipCoding" %}
-    case {{ variable.name }} = "{% call keyedNameFor type variable %}"
+  private enum CodableKeys: String, CodingKey {
+    {% for variable in type.instanceVariables where variable|!annotated:"skipCoding" %}
+    case {{ variable.name }} {% call keyedNameFor type variable %}
+    {% endfor %}
+  }
+{% endmacro %}
+
+{% macro inheritanceInitializer type %}
+  {% if type.supertype.based.AutoDecodable or type.supertype.based.Decodable %}
+    try self.init(from: decoder)
```

Comment:
Also this should use `decoder.superDecoder`

Reply:
When we're inheriting a type that is codable the initializer has to be convenience otherwise we can't override the original required initializer. At least thats what I found while working through this. I'll see if I can do it another way, however this seems to work without the superDecoder/encoder. For pure swift serializations using the superDecoder and superEncoder works, however, you end up with all of you super keys being at a different level of the final hash, i.e. inside a top level key `super`, which wouldn't be what you would want for api responses (imo).

Reply:
Yes, keys separation is not helpful in this case, but I still don't see how this will work with `let` properties.